### ISSUE TITLE
[BUG][multiple][OpenAPI-v3.1.0] SIMPLIFY_ONEOF_ANYOF is broken in v7.3.0,v7.4.0

### ISSUE THREAD

**Issue by guyarad** [OpenAPITools/openapi-generator#18184]
#### Bug Report Checklist

- [x] Have you provided a full/minimal spec to reproduce the issue?
- [x] Have you validated the input using an OpenAPI validator ([example](https://apidevtools.org/swagger-parser/online/))?
- [x] Have you [tested with the latest master](https://github.com/OpenAPITools/openapi-generator/wiki/FAQ#how-to-test-with-the-latest-master-of-openapi-generator) to confirm the issue still exists?
- [x] Have you searched for related issues/PRs?
- [x] What's the actual output vs expected output?
- [ ] [Optional] Sponsorship to speed up the bug fix or feature request ([example](https://github.com/OpenAPITools/openapi-generator/issues/6178))

##### Description

On v.7.3.0 (and  v7.4.0, the current latest release) the **simplify-oneof-anyof** normalizer was broken (at least for Python, but I believe for other generators as well). For optional model fields, an additional model is created which makes usage very cumbersome.

For example, consider an `Item` model with the following optional fields:
1. `my_enum` - optional enum of type `MyEnum`
2. `is_offer` - optional `bool`

The expected behavior is something like:
```python
class Item(BaseModel):
    is_offer: Optional[StrictBool] = None
    my_enum: Optional[MyEnum] = None
    ...
```

But  the actual behavior in the specified generator versions produced additional models and `Item` looks like:
```python
class Item(BaseModel):
    is_offer: Optional[IsOffer] = None
    my_enum: Optional[ItemMyEnum] = None
    ...
```

##### openapi-generator version

Bug appeared in v7.3.0, still in v7.4.0 and latest `master` (and didn't not exist in v7.2.0).

##### OpenAPI declaration file content or url

See full repro details in [gist](https://gist.github.com/guyarad/21bdd60b5643ac0d13ae5a57c593eefc#file-simplify_oneof_anyof-bug-md)

##### Generation Details

See full repro details in [gist](https://gist.github.com/guyarad/21bdd60b5643ac0d13ae5a57c593eefc#file-simplify_oneof_anyof-bug-md)

##### Steps to reproduce

See full repro details in [gist](https://gist.github.com/guyarad/21bdd60b5643ac0d13ae5a57c593eefc#file-simplify_oneof_anyof-bug-md)

##### Suggest a fix

**TL;DR** - the offending commit seems to be [this](https://github.com/OpenAPITools/openapi-generator/commit/301f25a3068db6dd17ab930f72975b235664f172) one. It changes the way `null` schema is identified, causing the simplification normalizer failing to do its job.

After cloning and using the "docker development" route I was able to use `git bisect` and each building+generating the client, identifying the commit quickly, so thank you for offering the docker packaging!

I believe the fix should include added tests for this behavior (not sure why existing tests did not identify it).

**Comment by guyarad** [comment]
Seems like the same behavior exist for other generators as well.

For example, in Java (notice the type of the parameters).

#### Optional fields in v7.2.0

```java
  public void setIsOffer(Boolean isOffer) {
    this.isOffer = isOffer;
  }

  public Item myEnum(MyEnum myEnum) {
    this.myEnum = myEnum;
    return this;
  }
```

#### Optional fields in v7.3.0

```java
  public void setIsOffer(IsOffer isOffer) {
    this.isOffer = isOffer;
  }

  public Item myEnum(ItemMyEnum myEnum) {
    this.myEnum = myEnum;
    return this;
  }
```

**Comment by wing328** [comment]
thanks for the details. i will try to fix it and add a test to ensure the issue is covered moving forward.

**Comment by wing328** [comment]
update: filed https://github.com/OpenAPITools/openapi-generator/pull/18189 with a test to fix the issue

**Comment by sidpan1** [comment]
Thanks @wing328 I tried your fix - it solves this particular issue. But it ignores the enum type for another related use case where an optional is used with a list.

```
# Actual pydantic class
class SomeModel(BaseModel):
    inputs: Optional[List[str]]

# Actual generated model (before this change)
class SomeModel(BaseModel):
    inputs: Optional[Inputs] = None

# Actual generated model (after this change - note that Inputs type has been removed)
class SomeModel(BaseModel):
    inputs: Optional[Any] = None

# Expected model
class SomeModel(BaseModel):
    inputs: Optional[List[StrictStr]]


```

**Comment by wing328** [comment]
@sidpan1 can you please share a spec to reproduce the issue?
